server/lib/two-factor-authentication/lib.ts

```diff
@@ -184,12 +188,21 @@ async function validateRequest(
       },
     });
 
+    const supportedMethods = await twoFactorMethodsSupportedByUser(remoteUser);
+    const authenticationOptions: Partial<Record<TwoFactorMethod, unknown>> = {};
+    for (const method of supportedMethods) {
+      if (typeof providers[method].authenticationOptions === 'function') {
+        authenticationOptions[method] = await providers[method].authenticationOptions(remoteUser, req);
+      }
+    }
```

Comment:
Are we supposed to ignore if authenticationOptions is not a function? If we're always expecting a function, it would be simpler to presume this is always a function and be strict about types here.